### ISSUE TITLE
Update anchor tags which don't have an href. 

### ISSUE THREAD

**Issue by Mike-Heneghan** [Mike-Heneghan/ALISS#108]
Currently, in some parts of the project, there are anchor tags which do not have an `href` attribute. This was noticed on the `Actions` tab with users being unable to select them using their keyboard only.

**Comment by Mike-Heneghan** [comment]
Updated the anchor tags for modals and other actions. 

Although the solution seemed to be successful the navigation to input fields when the button revealed a form input was a bit jarring. 

<img width="369" alt="Screenshot 2019-09-23 at 18 33 11" src="https://user-images.githubusercontent.com/36415632/65448415-a0dac880-de30-11e9-9d1a-f91bcf5bb37e.png">

Also, there was an occasion it seemed that the additional params might have confused the 'Back to last search" although I couldn't repeat it.

**Comment by Mike-Heneghan** [comment]
Noticed that the "Delete Service" modal "X close" and "Cancel" can't be reached with the keyboard. 

Also, the modal is incorrectly labelled 'Delete Organisation".

**Comment by Mike-Heneghan** [comment]
Began updating the links to add `href="#main"` stops the issues with the page navigating to items in an unexpected way.

**Comment by Mike-Heneghan** [comment]
The heading which works as a toggle for the actions box cannot be navigated to by keyboard. 

Added a link wrapper to allow it to be selected and operated when there is not enough space on the screen to have it rendered to the right of the content. 

<img width="142" alt="Screenshot 2019-09-27 at 11 03 55" src="https://user-images.githubusercontent.com/36415632/65761414-a37e3c00-e116-11e9-8fd7-89c410eaa45a.png">

Although when there is enough room the styling does cvhange and the link doesn't do anything:

<img width="417" alt="Screenshot 2019-09-27 at 11 00 56" src="https://user-images.githubusercontent.com/36415632/65761208-471b1c80-e116-11e9-8c34-b1d615b506e5.png">

The `href="#main"` may be a little hacky. I'd be curious if this is bad practice for screen readers etc.

**Comment by Mike-Heneghan** [comment]
https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element

**Comment by Mike-Heneghan** [comment]
Began updating the `toggle` method in JS to add `role="button"` and `aria-expanded` to inform users on a screen reader that something is being expanded and collapsed i.e. a form appears. Without this there is no feedback that 'clicking' these links does anything.

**Comment by Mike-Heneghan** [comment]
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role

**Comment by Mike-Heneghan** [comment]
https://www.smashingmagazine.com/2014/09/making-modal-windows-better-for-everyone/

**Comment by Mike-Heneghan** [comment]
https://bitsofco.de/accessible-modal-dialog/

**Comment by Mike-Heneghan** [comment]
I think the accordion-style click to expand form below for "Email this listing" for example should be using the aria-expanded and the aria-controls attributes as per https://www.w3.org/WAI/PF/aria-1.1/states_and_properties#aria-expanded

**Comment by Mike-Heneghan** [comment]
The solutions required based on the research above became a bit sprawling. Decided to split into more focussed issues.

**Comment by Mike-Heneghan** [comment]
Associated issues closed.
